coding.go

```diff
@@ -1,62 +1,18 @@
 package format
 
 import (
-	"fmt"
-	"sync"
-
 	blocks "github.com/ipfs/go-block-format"
 )
 
 // DecodeBlockFunc functions decode blocks into nodes.
 type DecodeBlockFunc func(block blocks.Block) (Node, error)
 
-type BlockDecoder interface {
-	Register(codec uint64, decoder DecodeBlockFunc)
-	Decode(blocks.Block) (Node, error)
-}
-type safeBlockDecoder struct {
-	// Can be replaced with an RCU if necessary.
-	lock     sync.RWMutex
-	decoders map[uint64]DecodeBlockFunc
-}
-
-// Register registers decoder for all blocks with the passed codec.
-//
-// This will silently replace any existing registered block decoders.
-func (d *safeBlockDecoder) Register(codec uint64, decoder DecodeBlockFunc) {
-	d.lock.Lock()
-	defer d.lock.Unlock()
-	d.decoders[codec] = decoder
-}
-
-func (d *safeBlockDecoder) Decode(block blocks.Block) (Node, error) {
+// Decode decodes the given block using the default BlockDecoder.
+func Decode(block blocks.Block, decoder DecodeBlockFunc) (Node, error) {
```

Comment:
Yeah, I'm cool just deleting this given we're breaking (a very small number of) users here anyway.

Reply:
The only thing this does is shortcut if your block is already a node which is not necessarily an obvious optimization. This makes the modifications upstream a no-brainer one line thing (import the codec + pass it) rather than a few lines of copy-paste.

Reply:
i'm okay with removing it entirely